src/WsHandler.js

```diff
@@ -0,0 +1,183 @@
+            WebSocket.createWebSocketStream(client),
+            JSONStream.parse(),
+            demuxSink,
+            err => {
+                if (err) {
+                    // A pipeline can multiplex multiple sessions however if one fails
+                    // the whole pipeline does as well,
+                    PromCollector.sessionErrorCount.inc();
+
+                    logger.error('[WsHandler] Connection pipeline: %o;  error: %o', connectionInfo, err);
+                }
+            });
+
+        connectionPipeline.on('finish', () => {
+            logger.info('[WsHandler] Connection pipeline successfully finished %o', connectionInfo);
+
+            // We need to explicity close the ws, you might notice that we don't do the same in case of an error
+            // that's because in that case the error will propagate up the pipeline chain and the ws stream will also
+            // close the ws.
+            client.close();
+        });
+
+        logger.info(
+            '[WsHandler] New app connected: ua: %s, protocol: %s, referer: %s',
+            ua,
+            client.protocol,
+            referer
+        );
+
+        client.on('error', e => {
+            logger.error('[WsHandler] Websocket error: %s', e);
+            PromCollector.connectionError.inc();
+        });
+
+        client.on('close', () => {
+            PromCollector.connected.dec();
+        });
+    }
+
+    /**
+     *
+     */
+    _createDemuxSink(connectionInfo) {
+        const demuxSinkOptions = {
+            connectionInfo,
+            dumpFolder: './temp',
+            log: logger
+        };
+
+        return new DemuxSink(demuxSinkOptions);
+    }
+
+    /**
+     *
+     */
+    _createClientMessageHandler(statsSessionId, demuxSink, client) {
+        const clientMessageHandlerOptions = {
+            statsSessionId,
+            tempPath: this.tempPath,
+            sequenceNumberSendingInterval: this.sequenceNumberSendingInterval,
+            demuxSink,
+            client
+        };
+
+        return new ClientMessageHandler(clientMessageHandlerOptions);
+    }
+
+    /**
+     *
+     * @returns
+     */
+    _createConnectionInfo(upgradeReq, referer, ua, client) {
+        // During feature extraction we need information about the browser in order to decide which algorithms use.
+        const connectionInfo = {
+            path: upgradeReq.url,
+            origin: upgradeReq.headers.origin,
+            url: referer,
+            userAgent: ua,
+            clientProtocol: client.protocol
+        };
+
+        connectionInfo.statsFormat = getStatsFormat(connectionInfo);
+
+        return connectionInfo;
+    }
+
+    /**
+     * Clear the connection timeout if the user is reconnected/
+     *
+     * @param {*} id
+     */
+    _clearConnectionTimeout(id) {
+        const timeoutId = this.sessionIdTimeouts[id];
+
+        if (timeoutId) {
+            logger.info('[WsHandler] Clear timeout for connectionId: %s', id);
```

Comment:
I'm assuming this means that the client reconnected, maybe we want to make that obvious in the log message and also add a prom metric so we can track how many reconnects happen.